tests/datastreams/test_public_api.py

```diff
@@ -30,3 +30,19 @@ def test_public_api():
                 sorted(["direction:in", "manual_checkpoint:true", "type:kinesis", "topic:stream-123"]), parent_hash
             )
             assert got.hash == expected
+
+
+def test_manual_checkpoint_behavior():
+    headers = {}
+    mocked_tracer = MockedTracer()
+    with mock.patch("ddtrace.tracer", new=mocked_tracer):
+        with mock.patch("ddtrace.config", new=MockedConfig()):
+            with mock.patch.object(mocked_tracer.data_streams_processor, "set_checkpoint") as mock_set_checkpoint:
+                set_consume_checkpoint("kinesis", "stream-123", headers.get)
+                called_tags = mock_set_checkpoint.call_args[0][0]
+                assert "manual_checkpoint:true" in called_tags
```

Comment:
Can we also test got.hash (similar to above) and make sure `manual_checkpoint:true` is / is not in the tags needed to compute the hash?